platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/ivyservice/projectmodule/LocalComponentRegistry.java

```diff
@@ -28,7 +26,6 @@
  * or in another build within a composite.
  */
 @ThreadSafe
-@ServiceScope(Scopes.Build.class)
```

Comment:
💭  Are we missing a scope for these kinds of services? It looks like they're in a separate hierarchy ("dependency resolution scope")

Reply:
Yeah the dependency resolution scope is weird. It has a bunch of stuff we need for performing dependency resolution, and gets instantiated either as a child of the project scope or a child of the build scoped services.

Its in an odd place, and perhaps it makes sense to discuss how/if we want to reposition those services in the hierarchy. 

I think the goal was that you should not need a project to perform resolution, and that also it is nice to separate the dependency resolution services from all other services -- for organization's sake. 

I've felt I wanted something similar for the Java plugins -- to have a separate set of services for the plugins that aren't accessible to (for example) the native plugins, or external plugins